Define cloud billing event types and codes

kimlisa opened the pull request:
part of https://github.com/gravitational/cloud/issues/442

#### Description
- Define cloud billing event types and codes
- Exported auth.clientUsername so it can be used in enterprise when emitting cloud related events (https://github.com/gravitational/teleport.e/pull/239)

kimlisa committed:
Define cloud billing event types and codes
- Exported auth.clientUsername so it can be used in enterprise

alex-kovoy commented on lib/events/codes.go:
```diff
@@ -294,6 +294,14 @@ const (
 	// MFADeviceDeleteEventCode is an event code for users deleting MFA devices.
 	MFADeviceDeleteEventCode = "T1007I"
+	// BillingCardCreateCode is emitted when a user creates a new credit card.
+	BillingCardCreateCode = "T1008I"
```
If you look at numbers, the `T10xxx` is used for user login events. The billing deserve own group. 

Lets use `TBL` prefix for that:

`TBL01I`
`TBL02I`

kimlisa replied:
revised

kimlisa committed:
Address CR
- Stayed with convention and created seperate
create(upsert) and delete event objects in proto file
- Renamed event code numbers

alex-kovoy commented on api/types/events/events.proto:
```diff
@@ -1263,6 +1297,9 @@ message OneOf {
         events.SessionUpload SessionUpload = 43;
         events.MFADeviceAdd MFADeviceAdd = 44;
         events.MFADeviceDelete MFADeviceDelete = 45;
+        events.BillingAccountUpdate BillingAccountUpdate = 46;
```
`BillingAcountUpdate` -> `BillingInformationUpdate`

alex-kovoy commented on lib/events/api.go:
```diff
@@ -192,6 +192,15 @@ const (
+	// BillingCardUpdateEvent is emitted when a user updates an existing credit card.
+	BillingCardUpdateEvent = "billing.update_card"
+	// BillingAccountUpdateEvent is emitted when a user updates their billing account.
+	BillingAccountUpdateEvent = "billing.update_account"
```
BillingAccountUpdate -> BillingInformationUpdate

kimlisa replied:
revised

alex-kovoy reviewed (approved):
with comments

kimlisa committed:
Change BillingAccount to BillingInformation

russjones reviewed (approved):
Bot.

kimlisa committed:
Export clientImpersonator for use in e
